src/test/java/com/ericsson/ei/controller/TestFreeStyleQueryImpl.java

```diff
@@ -0,0 +1,130 @@
+    private static String input;
+    private static final String DB_NAME = "MissedNotification";
+    private static final String DB_COLLECTION = "Notification";
+
+    @Value("${server.port}")
+    private String serverPort;
+
+    @Autowired
+    private ProcessQueryParams unitUnderTest;
+
+    @Before
+    public void setUp() {
+        input = FileUtils.readFileAsString(new File(inputPath));
+        try (MongoClient mongoClient = new MongoClient()) {
+            MongoDatabase db = mongoClient.getDatabase(DB_NAME);
+            com.mongodb.client.MongoCollection<Document> collection = db.getCollection(DB_COLLECTION);
+            final Document dbObjectInput = Document.parse(input);
+            collection.insertOne(dbObjectInput);
+            if (collection.count() != 0) {
+                System.out.println("Data Inserted successfully in both the Collections");
+            }
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
+        }
+    }
+
+    @Test
+    public void filterFormParamTest() throws Exception {
+        JSONObject inputJArr = new JSONObject(input);
+        LOGGER.info("The input string is : " + inputJArr.toString());
+        DB db = new MongoClient().getDB(DB_NAME);
+        Jongo jongo = new Jongo(db);
+        MongoCollection aggObjects = jongo.getCollection(DB_COLLECTION);
+        Document one = aggObjects.findOne().as(Document.class);
+        JSONObject json = new JSONObject(one.toJson());
+        LOGGER.info("Expect Output for FilterFormParamTest : " + json.toString());
+        System.out.println(json.toString());
+        JSONObject output = null;
+        try {
+            JsonNode inputCriteria = new ObjectMapper().readTree(QUERY);
+            JSONArray result = unitUnderTest.filterFormParam(inputCriteria);
+            output = result.getJSONObject(0);
+            LOGGER.info("Output for FilterFormParamTest is : " + output.toString());
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        assertNotNull(output);
+    }
+
+    @Test
+    public void filterQueryParamTest() throws JSONException {
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        request.setServerName("localhost:" + serverPort);
+        request.setRequestURI("/query/free?request=");
+        request.setQueryString("testCaseExecutions.testCase.verdict:PASSED,testCaseExecutions.testCase.id:TC5,id:6acc3c87-75e0-4b6d-88f5-b1a5d4e62b43");
+        String url = request.getRequestURL() + request.getQueryString();
+        DB db = new MongoClient().getDB(DB_NAME);
+        Jongo jongo = new Jongo(db);
+        MongoCollection aggObjects = jongo.getCollection(DB_COLLECTION);
+        Document one = aggObjects.findOne().as(Document.class);
+        JSONObject json = new JSONObject(one.toJson());
```

Comment:
same applies here for json local variable. It is not used in any assertion hence remove it.